internal/app/triggermessageprocessor.go

```diff
@@ -0,0 +1,143 @@
+
+import (
+	"fmt"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/appfunction"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/common"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/runtime"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/trigger"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/bootstrap/messaging"
+	"github.com/edgexfoundry/go-mod-bootstrap/v2/di"
+	"github.com/edgexfoundry/go-mod-messaging/v2/pkg/types"
+	"github.com/hashicorp/go-multierror"
+	"sync"
+)
+
+type simpleTriggerServiceBinding struct {
+	*Service
+	*runtime.GolangRuntime
+}
+
+func (b *simpleTriggerServiceBinding) SecretProvider() messaging.SecretDataProvider {
+	return container.SecretProviderFrom(b.Service.dic.Get)
+}
+
+func NewTriggerServiceBinding(svc *Service) trigger.ServiceBinding {
+	return &simpleTriggerServiceBinding{
+		svc,
+		svc.runtime,
+	}
+}
+
+func (b *simpleTriggerServiceBinding) DIC() *di.Container {
+	return b.Service.dic
+}
+
+func (b *simpleTriggerServiceBinding) BuildContext(env types.MessageEnvelope) interfaces.AppFunctionContext {
+	return appfunction.NewContext(env.CorrelationID, b.Service.dic, env.ContentType)
+}
+
+func (b *simpleTriggerServiceBinding) Config() *common.ConfigurationStruct {
+	return b.config
+}
+
+// customTriggerBinding wraps the CustomTriggerServiceBinding interface so that we can attach methods
+type triggerMessageProcessor struct {
+	bnd trigger.ServiceBinding
+}
+
+// Process provides runtime orchestration to pass the envelope / context to the pipeline.
+// Deprecated: This does NOT support multi-pipeline usage.  Will send a message to the default pipeline ONLY and throw if not configured.  Use MessageReceived.
+func (mp *triggerMessageProcessor) Process(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope) error {
+	context, ok := ctx.(*appfunction.Context)
+	if !ok {
+		return fmt.Errorf("App Context must be an instance of internal appfunction.Context. Use NewAppContext to create instance.")
+	}
+
+	defaultPipelines := mp.bnd.GetMatchingPipelines(interfaces.DefaultPipelineId)
+
+	if len(defaultPipelines) != 1 {
+		return fmt.Errorf("TriggerMessageProcessor does not support multiple pipelines.  Please use TriggerMessageHandler for multi-pipeline support")
+	}
+
+	messageError := mp.bnd.ProcessMessage(context, envelope, defaultPipelines[0])
+	if messageError != nil {
+		// ProcessMessage logs the error, so no need to log it here.
+		return messageError.Err
+	}
+
+	return nil
+}
+
+// MessageReceived provides runtime orchestration to pass the envelope / context to configured pipeline(s).
+func (mp *triggerMessageProcessor) MessageReceived(ctx interfaces.AppFunctionContext, envelope types.MessageEnvelope, outputHandler interfaces.PipelineResponseHandler) error {
+	return mp.messageReceived(ctx, envelope, outputHandler)
```

Comment:
Why need to use internal func rather than just having the code in this func?

Reply:
good catch this must be a refactoring artifact.